cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/kafka/LoggingEventSerializer.java

```diff
@@ -77,6 +78,36 @@ public ILoggingEvent fromBytes(ByteBuffer buffer) throws IOException {
   }
 
   /**
+   * Decodes the timestamp of a {@link ILoggingEvent} encoded in the given {@link ByteBuffer} with the schema
+   * returned by the {@link #getAvroSchema()} method.
+   *
+   * @param buffer the buffer to decoder
+   * @return the event timestamp
+   * @throws IOException if fail to decode
+   */
+  public long decodeEventTimestamp(ByteBuffer buffer) throws IOException {
```

Comment:
where is it being used? In that context, would it make sense to throw a different exception if the timestamp fields cannot be found?